zipkin_thrift_span_test.go

```diff
@@ -310,6 +310,35 @@ func TestBaggageLogs(t *testing.T) {
 	assert.NotNil(t, findAnnotation(thriftSpan, `{"event":"baggage","key":"auth.token","value":"token"}`))
 }
 
+func TestMaxAnnotationLength(t *testing.T) {
+	value := make([]byte, 512)
+	tests := []struct {
+		tagValueLength int
+		value          []byte
+		expected       []byte
+	}{
+		{256, value, value[:256]},
+		{512, value, value},
+	}
+
+	for _, test := range tests {
+		func() {
```

Comment:
why is this func needed? is it for the defer? If so, can we replace with `t.Run` instead?